cachi2/core/package_managers/generic/models.py

```diff
@@ -29,9 +42,51 @@ def no_empty_checksums(cls, value: dict[str, str]) -> dict[str, str]:
             raise ValueError("At least one checksum must be provided.")
         return value
 
+    @model_validator(mode="after")
+    def set_target(self, info: ValidationInfo) -> "LockfileArtifact":
+        """Set the target path if not provided and resolve it into an absolute path."""
+        if not self.target:
+            url_path = urlparse(str(self.download_url)).path
+            self.target = Path(url_path).name
+
+        # needs context
+        if not info.context or "output_dir" not in info.context:
```

Comment:
I think that tying the class instantation to the existence of specific data in context is not ideal, but the alternative would be to either do the `target` validation within the `_resolve_generic_lockfile` (while also downloading the files), or even make the class mutable if we happen to need this info further down the line. 

So I'm fine with the current approach.

Reply:
Thanks. There was a couple of options, but based on previous review comments, this seems to be the preferred option. I'm aware of the downsides that come with it.

Reply:
The unfortunate reality here is that it's a nested model, so to IMO do it by the book would mean to effectively validate the lockfile twice - once as a dict from the YAML parser, loop over all artifacts and override any missing targets and second via models. The former however would be relying on the structure which the pydantic models validate so we can't use that approach sadly, though contexts aren't the cleanest way, I agree (but hey, they showcase a similar thing in the docs!)

> alternative would be to either do the target validation within the _resolve_generic_lockfile

Doable and also kinda clean to be honest :), I'll leave it to @kosciCZ to decide.

> or even make the class mutable if we happen to need this info further down the line.

Honestly I'd rather keep using contexts for validators like this than go with mutable models, to me, these are the sources of truth for the rest of the logic so changing them may introduce hard to catch issues later on.

@kosciCZ depending on which approach you stick to, can you please update the commentary to be more verbose on the context need?

Reply:
Update the comment to be more verbose